pubsub/iterator.go

```diff
@@ -123,12 +132,16 @@ func newMessageIterator(subc *vkit.SubscriberClient, subName string, po *pullOpt
 	pingTicker := time.NewTicker(30 * time.Second)
 	cctx, cancel := context.WithCancel(context.Background())
 	cctx = withSubscriptionKey(cctx, subName)
+
+	subID := strings.Split(subName, "/")[3]
```

Comment:
Two questions:

* Why normalize the subscription ID at all?  Are there concerns about metric collisions because the subscription won't be fully qualified?

* Is this expectation safe, or are there cases where we might have atypical subscription IDs?  Is `len(a)-1` a better idx value to always grab the last element?

Reply:
1. SubscriptionID is preferred over the fully qualified name for ease of viewing in traces. For example, the span name "otel-subscription process" shows up nicer than "projects/my-project/subscriptions/otel-subscription process" in Cloud Trace. For metrics, we plan to use the fully qualified name.
2. It probably doesn't hurt to use your suggestion, though I'll probably abstract it away in a helper function. Thanks for the suggestion.